ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -236,26 +268,47 @@ std::shared_ptr<Function> PrePostProcessor::build(const std::shared_ptr<Function
                 new_param_shape[width_idx] = input->m_tensor_data->get_spatial_width();
             }
         }
-        auto new_param = std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), new_param_shape);
-        if (input->m_tensor_data->is_layout_set()) {
-            new_param->set_layout(input->m_tensor_data->get_layout());
-        }
-        // Old param will be removed, so friendly name can be reused
-        new_param->set_friendly_name(param->get_friendly_name());
 
-        // Also reuse names of original tensor
-        new_param->get_output_tensor(0).set_names(param->get_output_tensor(0).get_names());
+        std::vector<std::shared_ptr<ov::Node>> nodes;
+        std::vector<std::shared_ptr<op::v0::Parameter>> new_params;
+
+        // Create separate parameter for each plane. Shape and friendly name is based on color format
+        for (size_t plane = 0; plane < color_info->planes_count(); plane++) {
+            auto plane_shape = color_info->shape(plane, new_param_shape);
+            auto plane_param =
+                std::make_shared<op::v0::Parameter>(input->m_tensor_data->get_element_type(), plane_shape);
+            if (plane < input->m_tensor_data->planes_sub_names().size()) {
+                auto sub_name = std::string("/") + input->m_tensor_data->planes_sub_names()[plane];
+                inherit_friendly_names(param, plane_param, sub_name);
+            } else {
+                auto sub_name = color_info->friendly_suffix(plane);
+                inherit_friendly_names(param, plane_param, sub_name);
+            }
+            if (input->m_tensor_data->get_layout() != Layout()) {
+                plane_param->set_layout(input->m_tensor_data->get_layout());
+            }
+            new_params.push_back(plane_param);
+            nodes.push_back(plane_param);
+        }
 
-        std::shared_ptr<Node> node = new_param;
-        PreprocessingContext context(new_param->get_layout());
+        PreprocessingContext context(input->m_tensor_data->get_layout());
+        context.color_format() = input->m_tensor_data->get_color_format();
         context.network_layout() = param->get_layout();
         context.network_shape() = param->get_partial_shape();
+
         // 2. Apply preprocessing
         for (const auto& action : input->m_preprocess->actions()) {
-            node = std::get<0>(action)({node}, context);
+            auto node = std::get<0>(action)(nodes, context);
+            nodes = {node};
             tensor_data_updated |= std::get<1>(action);
         }
 
+        OPENVINO_ASSERT(nodes.size() == 1,
+                        "Multiple plane input is not allowed as network input. Consider using of convert_color "
+                        "preprocessing operation");
+        OPENVINO_ASSERT(is_rgb_family(context.color_format()) || context.color_format() == ColorFormat::UNDEFINED,
+                        "Final input color format is incorrect");
```

Comment:
can we extend error message to be more detailed - e.g. NV12 conversion to RGB / BGR is only supported

Reply:
Done